client/my-sites/plugins/plugin-details-CTA/index.jsx

```diff
@@ -68,7 +68,10 @@ const PluginDetailsCTA = ( {
 		isEligibleForAutomatedTransfer( state, selectedSite?.ID )
 	);
 	const hasEligibilityMessages =
-		! isAtomic && ! isJetpack && ( eligibilityHolds || eligibilityWarnings || isEligible );
+		! isAtomic &&
+		! isJetpack &&
+		isEligible &&
```

Comment:
`isEligible` means `isEligibleForAutomatedTransfer` which isn't the purpose of the `hasEligibilityMessages` check. For sites having `isEligible = false` (it happens to my free site), I don't get the modal and I am redirected to checkout directly. I believe it was mistakenly added there and it worked so long cause it was part of an or `||`. Removing `isEligible` seems working as expected for me.

Your change works fine with a biz or ecom plan with domain, redirecting to checkout directly without showing an empty modal.

Reply:
Good catch! I will remove `isEligible` from the condition and I will test the following scenarios:
- Business plan without custom domain
- Business plan with custom domain
- Free plan without custom domain
- Free plan with custom domain

Please let me know if you can think of any other scenarios to test.

Reply:
They seem enough